[CYS] Update the intro page

albarin opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/woocommerce/woocommerce/issues/46837

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**If the currently active block theme was already customized, the intro page design should be updated**
1. Go to `wp-admin/tools.php?page=woocommerce-admin-test-helper` click on `Reset Customize Your Store`.
2. Go to `Appearance > Theme` and select the TT4 theme.
3. Go to `WooCommerce > Home`, click on `Customize your store`, then on `Start designing`.
4. Go again to `WooCommerce > Home`, click on `Customize your store`.
5. Make sure the intro page looks like this 👇 
<img width="1486" alt="Screenshot 2024-04-30 at 16 33 38" src="https://github.com/woocommerce/woocommerce/assets/186112/6cf01450-4092-4fa9-ad87-5f4d9f42e3d1">

6. Click on `Customize your theme` and check you are redirected to the assembler.
7. Go to `Appearance > Theme` and select a classic theme (like Storefront or TT1).
8. Go again to `WooCommerce > Home`, click on `Customize your store` and `Customize your theme`.
9. Make sure you are now redirected to the Customizer (not the assembler).
10. Go again to `WooCommerce > Home`, click on `Customize your store`.
11. Click on `Browse themes` and make sure you are redirected to `WooCommerce → Extensions → Themes`.
12. Go again to `WooCommerce > Home`, click on `Customize your store`.
13. Click on `Use the store designer` and make sure you are redirected to the Customizer.
14. Go to `Appearance > Theme` and select the TT4 theme.
15. Go again to `WooCommerce > Home`, click on `Customize your store`.
16. Click on `Use the store designer` and make sure you are redirected to the assembler.

**Trying to access the assembler with a different theme than TT4 should ask the user to design a new theme**
1. Go to `wp-admin/tools.php?page=woocommerce-admin-test-helper` and click on `Reset Customize Your Store`.
2. Go to `Appearance > Theme` and select a block theme (other than TT4, like Tsubaki or TT3).
3. Go to `WooCommerce > Home`, click on `Customize your store`, then on `Start designing`.
4. Make sure you see the following modal:
<img width="551" alt="Screenshot 2024-06-05 at 10 02 57" src="https://github.com/woocommerce/woocommerce/assets/186112/638a008e-475a-407e-84ef-a78dc9a36b17">


<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
[CYS] Remove the restriction to TT4 and allow users to proceed to the pattern assembler with any block themes. Update intro page design.
#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

albarin committed:
Change the modal condition to check if the theme is a block theme

albarin committed:
Only switch to TT4 if the theme is not a block theme

albarin committed:
Remove unnecessary code

albarin committed:
Add new Intro cards when the theme is customized

albarin committed:
Redirect to the customizer if the active theme is a classic one

albarin committed:
Update the "Start designing" link depending on the type of theme

albarin committed:
Fix lint errors

albarin committed:
Fix more lint errors

albarin committed:
Fix tests

albarin committed:
Remove only

albarin committed:
Add tests for the customize button

albarin committed:
Update copy

albarin committed:
Fix lint errors

albarin committed:
Try to fix tests

albarin committed:
Fix tests

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

albarin commented on plugins/woocommerce-admin/client/customize-store/intro/intro-banners.tsx:
```diff
@@ -331,14 +341,21 @@ export const ExistingNoAiThemeBanner = () => {
 			buttonIsLink={ false }
 			bannerButtonOnClick={ () => {
 				trackEvent( 'customize_your_store_intro_customize_click' );
```
@nefeline do you know if we should keep this event or maybe send a different one depending on the type of theme?
Also, do we need any other events for this new design links? TY!

nefeline replied:
> do you know if we should keep this event or maybe send a different one depending on the type of theme?

I don't 🤔 : thinking about this, perhaps we should preserve `customize_your_store_intro_customize_click` as-is and add extra events to check for the theme types. Maybe we could check with Veronica as well? IMO the more information we can collect, the better!

nefeline replied:
> Also, do we need any other events for this new design links? TY!

Yes! We should def. add tracking events to the two new cards:

<img width="903" alt="Screenshot 2024-05-02 at 12 36 15" src="https://github.com/woocommerce/woocommerce/assets/15730971/c2ec2d68-2929-4465-9e2e-4c239a7ef464">

albarin replied:
I have added events to those 2 new links and added a property with the type of theme to the old `customize_your_store_intro_customize_click`.

https://github.com/woocommerce/woocommerce/pull/46916/commits/2c7be044aa4288100fd5c36eaaab8f37ea5895b5

I'll double check in the channel with Veronica next week when she's back.

albarin committed:
Remove commented out css code

albarin committed:
Fix tests

albarin committed:
Remove unused

nefeline reviewed (commented):
Thank you so much for working on this @albarin ! Sharing here my test results:

## With the TT2 theme:

* I was able to access the Assembler normally and change all templates as expected.
* After finishing my customizations and heading to the frontend, I get the following errors:

<img width="1230" alt="Screenshot 2024-05-02 at 12 14 21" src="https://github.com/woocommerce/woocommerce/assets/15730971/57410545-5b44-42d0-9f85-e416eef108aa">

<img width="1238" alt="Screenshot 2024-05-02 at 12 14 30" src="https://github.com/woocommerce/woocommerce/assets/15730971/bf76b49d-24b4-411d-9008-e33db0f63a2f">

we might have to open an issue to investigate why these template parts are being deleted.

## When switching from TT2 to TT4 theme

* I noticed that I can't access the assembler anymore after switching my current active theme from TT2 to TT4: the intro screen is just refreshed.

https://github.com/woocommerce/woocommerce/assets/15730971/588c411b-6732-4531-9bb3-c496862c3455

* If I reset the CYS then I'm presented with the initial "Design your own" screen again and I can access the assembler

Considering it is a very common and possible scenario for a user to switch between different themes, I think we should fix this and ensure the Assembler is accessible without the need to reset the CYS, WDYT?

# When using a classic theme

* ✅ Confirmed that I'm redirected to the customizer

## When having a customized block theme

* ✅ Confirmed that I'm redirected to the assembler

## The new cards:

Confirmed that they are both displayed and the links are correct and leading users to the expected path:


<img width="903" alt="Screenshot 2024-05-02 at 12 36 15" src="https://github.com/woocommerce/woocommerce/assets/15730971/bcad1900-3ae9-4431-b776-de283b984ccc">


Would it be worth adding a margin or padding below these two? Right now they are tied to the bottom of the screen.

albarin commented:
> After finishing my customizations and heading to the frontend, I get the following errors:

True! I'm not seeing the errors, but I don't see the header or footers 😱 and it happens with other block themes as well (TT3, Tsubaki, etc).

> I noticed that I can't access the assembler anymore after switching my current active theme from TT2 to TT4: the intro screen is just refreshed.

I could reproduce, thanks.

> Would it be worth adding a margin or padding below these two? Right now they are tied to the bottom of the screen.

Yes, makes sense.

I'll start working into all of these now, thanks for the review 🙏

albarin committed:
Add margin to the cards

gigitux commented:
> True! I'm not seeing the errors, but I don't see the header or footers 😱 and it happens with other block themes as well (TT3, Tsubaki, etc).
> 
> >

There is a dedicated issue for this: https://github.com/woocommerce/woocommerce/issues/42181. Luckly, it looks like that @Aljullu is working on the fix! ❤️

albarin committed:
Change active theme has mods by customize task completed

albarin committed:
Merge branch 'trunk' into 46837-remove-tt4-restriction

albarin committed:
Add events to the new intro page

albarin committed:
Fix color palette tests

albarin committed:
Remove unused 'fetchActiveThemeHasMods' function

This fixes the "select( 'core' ).getCurrentThemeGlobalStylesRevisions() is deprecated since version 6.5.0" warning

albarin commented:
@nefeline btw, this would get rid of the first deprecation warning (`getCurrentThemeGlobalStylesRevisions`) on this issue https://github.com/woocommerce/woocommerce/issues/47012

albarin committed:
Merge branch 'trunk' into 46837-remove-tt4-restriction

albarin committed:
Revert allowing any block theme in the assembler

albarin commented:
@gigitux @nefeline I've removed the ability to use any other theme apart from TT4 (context: p1717499442912829-slack-C053716F2H2). I've left most of the code to check the theme there though, since we are going to need it eventually. 
Could you please have another look at the PR? 🙏

gigitux commented:
Thanks @albarin for working on this! I followed the testing instructions, and everything worked okay except for steps 13 - 15.


In this video, I have Tsubaki as active theme. I'm not sure if this beavhiour is the right one:


https://github.com/woocommerce/woocommerce/assets/4463174/c9faf4a8-beec-4d43-bcc9-a0df243e44ff



If I click on `Customize your theme` and `Use the store designer,` I see that the Assembler is loaded with the Tsubaki default template.

Instead, in this case, I would:

- redirect to Site Editor when the user clicks on `Customize your theme`(it is necessary to update the label too. Currently, it is: `Continue to customize your store using the store designer. Change your color palette, fonts, page layouts, and more.`
- start the CYS setup flow when the user clicks on `Use the store designer`.

albarin committed:
Merge branch 'trunk' into 46837-remove-tt4-restriction

gigitux commented:
> @gigitux Thanks for the review. Sorry, yesterday I made a change to how this should work but steps 13-15 were not updated accordingly 🙏 I have updated them now, I've also added an extra test section as well to test the assembler cannot be accessed with any block theme which is not TT4.

No worries! Thanks for updating the testing instructions! In some specific conditions, I'm able to access the Assembler even if I'm not using TT4. If it is not super complex to fix it, I think that we should fix it.

- Start the CYS flow with TT4.
- Save.
- Switch to another block theme.
- Visit `wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store`
- The Assembler is accessible if I click on `Customize your theme` and `Use the store designer`.


https://github.com/woocommerce/woocommerce/assets/4463174/2b5dbdfa-3826-48a3-ba00-989c0d52f79b

albarin committed:
Prevent other block themes to go to the assembler

albarin commented:
@gigitux I've pushed a fixed, TY!

gigitux commented on plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js:
```diff
@@ -300,18 +299,6 @@ test.describe( 'Assembler -> Color Pickers', () => {
-
-			await saveButton.click();
-
-			await waitResponse;
```
Is there a particular reason for removing this?

gigitux commented on plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js:
```diff
@@ -300,18 +299,6 @@ test.describe( 'Assembler -> Color Pickers', () => {
-			await saveButton.click();
-
-			await waitResponse;
-
```
Is there a particular reason for removing this?

albarin replied:
No, not sure what happened, I've restored it and fixed the lint errors.

gigitux reviewed (approved):
Thanks for working on this! Everything LGTM now! Before merging this PR, can you:
- restore the changes that you did to the `color-picker` E2E test? It looks they are unrelated to this PR 🤔 
- fix lint errors

🚀 

Thanks!

nefeline reviewed (approved):
Thanks for your amazing work here, @albarin ! I've ran all tests and confirmed everything works as described. I just have a note regarding the following step with classic themes:

> Click on Use the store designer and make sure you are redirected to the Customizer.

Not at all related to your work here, but it feels a bit odd to have two links on the same page directing the user to the customizer. This is more of a UX concern & not related to the implementation, but probably worth considering a discussion whenever we start working with extensibility.

albarin committed:
Fix linting errors, restore test

albarin committed:
Merge branch 'trunk' into 46837-remove-tt4-restriction

gigitux commented on plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/color-picker.spec.js:
```diff
@@ -287,9 +287,9 @@ test.describe( 'Assembler -> Color Pickers', () => {
 		colorPalette
 	) ) {
-		test( `Color palette ${ colorPaletteName } should be applied`, async ( {
+		test.only( `Color palette ${ colorPaletteName } should be applied`, async ( {
```
I think that we should remove `only` too:
```suggestion
		test( `Color palette ${ colorPaletteName } should be applied`, async ( {
```

albarin committed:
Remove only from test

Co-authored-by: Luigi Teschio <gigitux@gmail.com>

albarin committed:
Fix lint error

jsnajdr commented:
Hi @albarin :wave: The `professional-theme.svg` image that this PR is adding is super big, 24MB. It's an SVG that embeds four PNG images, 4076x2426px, 6MB each:

<img width="400" alt="Screenshot 2024-06-18 at 11 46 31" src="https://github.com/woocommerce/woocommerce/assets/664258/fb172976-39b3-4b22-a4e7-a3a477942a38">

Let's optimize it. After all, it ends up being used as a 170x100px illustration for the "Choose a professionally designed theme" link.

albarin commented:
Sorry @jsnajdr, totally missed this. I've just opened a PR to optimize the image: https://github.com/woocommerce/woocommerce/pull/48566
